tests/manager/test_extended_async_sa_engine.py

```diff
@@ -0,0 +1,588 @@
+from __future__ import annotations
+
+from collections.abc import AsyncIterator
+from dataclasses import dataclass
+
+import pytest
+import sqlalchemy as sa
+from sqlalchemy.ext.asyncio import AsyncConnection
+from sqlalchemy.orm import declarative_base
+
+from ai.backend.manager.models.utils import ExtendedAsyncSAEngine
+
+
+@dataclass(frozen=True)
+class IsolationTestRowData:
+    """
+    Represents a test data row from the test_isolation_data table.
+    """
+
+    value: str
+    version: int
+
+
+Base = declarative_base()
+
+
+class IsolationDataModel(Base):  # type: ignore
+    """ORM model for test_isolation_data table."""
+
+    __tablename__ = "test_isolation_data"
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    value = sa.Column(sa.String(255), nullable=False)
+    version = sa.Column(sa.Integer, nullable=False, default=1)
+
+    def __init__(self, value: str, version: int = 1) -> None:
+        self.value = value
+        self.version = version
```

Comment:
It doesn't seem necessary to go to the trouble of creating an arbitrary test model to test; just create an isolation level session and test that.